File: frontend/src/types/accountUsage.ts
export type AccountUsagePeriod = "week" | "month";

export interface AccountUsageTotals {
  task_count: number;
  formal_task_count: number;
  trial_task_count: number;
  api_request_count: number;
  token_count: number;
  input_token_count: number;
  output_token_count: number;
}

export interface AccountUsagePeriodMetrics extends AccountUsageTotals {
  label: string;
  start_date: string;
  end_date: string;
}

export interface AccountUsageSummaryItem extends AccountUsageTotals {
  account_id: string;
}

export interface AccountUsageDetailResponse {
  account_id: string;
  period: AccountUsagePeriod;
  current: AccountUsagePeriodMetrics;
  history: AccountUsagePeriodMetrics[];
  all_time: AccountUsageTotals;
}

export interface AccountQuotaBalance {
  limit: number;
  used: number;
  remaining: number;
}

export interface CurrentAccountUsageResponse {
  account_id: string;
  period_start: string;
  period_end: string;
  usage: AccountUsageTotals;
  trial_api_request_balance: AccountQuotaBalance;
}

File: backend/app/schemas/account_usage.py
from __future__ import annotations

from datetime import date
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


class AccountUsageTotals(BaseModel):
    task_count: int
    formal_task_count: int
    trial_task_count: int
    api_request_count: int
    token_count: int
    input_token_count: int
    output_token_count: int


class AccountUsagePeriodMetrics(AccountUsageTotals):
    label: str
    start_date: date
    end_date: date


class AccountUsageSummaryItem(AccountUsageTotals):
    account_id: UUID


class AccountUsageDetailResponse(BaseModel):
    account_id: UUID
    period: Literal["week", "month"]
    current: AccountUsagePeriodMetrics
    history: list[AccountUsagePeriodMetrics]
    all_time: AccountUsageTotals


class AccountQuotaBalance(BaseModel):
    limit: int
    used: int
    remaining: int


class CurrentAccountUsageResponse(BaseModel):
    account_id: UUID
    period_start: date
    period_end: date
    usage: AccountUsageTotals
    trial_api_request_balance: AccountQuotaBalance
